### PR TITLE
Add open/closed field support for issue index

### PR DESCRIPTION
**Pull request by techknowlogick**
A couple of notes:
* Future changes should refactor arguments into a struct
* This filtering only is supported by meilisearch right now
* Issue index number is bumped which will cause a re-index
* Issue type should be added as well to support searching between issues & PRs

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Review by denyskon — APPROVED** [review]

File: modules/indexer/issues/meilisearch/meilisearch.go [hunk 1]
```diff
@@ -70,12 +70,19 @@ func (b *Indexer) Delete(_ context.Context, ids ...int64) error {
...
 		repoFilters = append(repoFilters, "repo_id = "+strconv.FormatInt(repoID, 10))
 	}
 	filter := strings.Join(repoFilters, " OR ")
+	if state == "open" || state == "closed" {
+		if filter != "" {
+			filter = "(" + filter + ") AND state = " + state
+		} else {
+			filter = "state = " + state
+		}
+	}
```
Comment (wxiaoguang) [inline modules/indexer/issues/meilisearch/meilisearch.go:85]: At the moment it is safe but there might be some changes cause SQL injection in the future for such code.

Comment (wxiaoguang) [reply modules/indexer/issues/meilisearch/meilisearch.go:85]: And, how could the code work if `var state="open"`?

Then the SQL becomes `WHERE state = open`, but it should be `WHERE state = 'open'` (quoted) ?

Comment (techknowlogick) [reply modules/indexer/issues/meilisearch/meilisearch.go:85]: This isn't sql, but instead a custom query language by meilisearch https://www.meilisearch.com/docs/reference/api/search#filter no quotes needed (but can be optionally used)

Comment (wxiaoguang) [reply modules/indexer/issues/meilisearch/meilisearch.go:85]: Ohhh, just realized that the file is `meilisearch.go`

**Comment by wolfogre** [comment]
> Future changes should refactor arguments into a struct

@techknowlogick I'm working on this, that's why I sent some splitted PRs to improve indexing. 😂

Don't worry, this PR won't cause too much conflict, but if you are still doing something like this PR, please let me know.

File: modules/indexer/issues/meilisearch/meilisearch.go (hunk 1 above)
Comment (techknowlogick) [reply modules/indexer/issues/meilisearch/meilisearch.go:85]: no worries ;)

**Comment by techknowlogick** [comment]
Blocking so it doesn't get merged before I add issue type as well.

**Pull request merged**